packages/fxa-payments-server/src/routes/Checkout/index.tsx

```diff
@@ -421,20 +429,22 @@ export const Checkout = ({
             </>
           </div>
         </div>
-        <PlanDetails
-          {...{
-            className: classNames('default', {
-              hidden: transactionInProgress && isMobile,
-            }),
-            selectedPlan,
-            isMobile,
-            showExpandButton: isMobile,
-          }}
-        />
-        {config.featureFlags.subscriptionCoupons ? (
-          // To be updated in issue #7097
-          <div data-testid="coupon-container"></div>
-        ) : null}
+        <CouponContext.Provider value={{ coupon, setCoupon }}>
```

Comment:
Could we change the Coupon component to use props instead of using Context? This seems to be the recommended approach and more closely matches what we do in the rest of the app.

Reply:
The PlanDetails and Coupon components are sibling elements. I'm not super fluent in React, but I believe the proper way to pass data between siblings is either by using a data store (like redux or whatever) or with the context/hooks API. I think you probably could get something going with properties but usually properties are for passing data down from a parent into a child component. There are a few other uses of the context API in the app and if I recall correctly, context/hooks is the way Bianca mentioned the app moving towards.

Reply:
Reference: https://www.pluralsight.com/guides/how-to-pass-data-between-react-components

Reply:
We've been passing functions through props.  For example, here's the checkout page passing functions to the new user form: https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/routes/Checkout/index.tsx#L314:L325

Reply:
The react docs are also a good reference for this.

https://reactjs.org/docs/context.html#before-you-use-context

Reply:
context removed